### PR TITLE
Issue 89 mtucheck

### PR DESCRIPTION
**Pull request by mortenjc**
### Issue reference / description

Added checks for MTU (ethernet packet sizes accepted by the 
link layer)

Default is now to expect 9000 bytes on all active interfaces on the server where EFU
is run. This currently broke the "make runefu" target which has been modified. It will likely also break the integration test, but is easily fixed. Has been tested and works on CentOS and MacOS.

Not unit tests have been created - this relies on ioctl() and other system calls

## Checklist for submitter

- [x] Check for conflict with integration test
- [x] Unit tests pass

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [x] Recommend for group code review

Also, nominate it on the code_review Slack channel.

### CODE REVIEW

File: prototype2/efu/HwCheck.cpp [hunk 1]
```diff
@@ -93,6 +93,7 @@ void HwCheck::debugPrint(struct ifaddrs * ifa) {
 }
 
 
+#if 0
 int main(int argc, char * argv []) {
```
Comment (SkyToGround) [inline prototype2/efu/HwCheck.cpp]: Seeing as this code is basically commented out, I think it should be removed. One alternative is to add it to a unit test.

File: prototype2/efu/HwCheck.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <net/if.h>
+
+class HwCheck {
+public:
```
Comment (SkyToGround) [inline prototype2/efu/HwCheck.h]: In my opinion, having this be a C-function is better than having it be a C++ class. The reason being that there is very little state that is kept track of and the member function is only called once.

Comment (mortenjc) [reply prototype2/efu/HwCheck.h]: I expect this class to evolve soon to contain more hardware related checks and hence more member functions.

File: prototype2/efu/HwCheck.cpp (hunk 1 above)
Comment (mortenjc) [reply prototype2/efu/HwCheck.cpp]: yes, I forgot to delete the code. It started out as a standalone program for testing

File: prototype2/efu/HwCheck.h (hunk 2 above)
Comment (SkyToGround) [reply prototype2/efu/HwCheck.h]: The question is still: do you expect the member functions to share state and do you expect that they will be called more than once per application run?

**Pull request merged**